### PR TITLE
Workaround non-null parent requirement of StubBase

### PR DESCRIPTION
**Pull request by jsjeon**

### CODE REVIEW

**Review by liutikas — COMMENTED** [review]
Clever workaround!

**Comment by jsjeon** [comment]
cc @hick209 as per https://github.com/facebook/ktfmt/issues/495 history

File: core/pom.xml [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
     <properties>
         <dokka.version>0.10.1</dokka.version>
-        <kotlin.version>1.8.22</kotlin.version>
+        <kotlin.version>2.0.20</kotlin.version>
```
Comment (hick209) [inline core/pom.xml:19]: If we don't touch this here would things still work?
If not, do you know what would be the min version we could put here?

**Review by hick209 — COMMENTED** [review]
I really appreciate the code you put here, but I wonder if we need to update the Kotlin version.

My worry is that if we put Kotlin 2.0.20 here then maybe people trying to use ktfmt with an earlier version of Kotlin will have issues integrating with it.
I understand that running ktfmt from a binary should not be problem, but there are instances where it is used as a code integration, like in LSPs or running it as a deamon.

File: core/pom.xml (hunk 1 above)
Comment (jsjeon) [reply core/pom.xml]: Not necessary to update the version; it was just handy to reproduce the issue. Back to the old version. :)

**Comment by hick209** [comment]
nit: would you mind adding the change to the [`CHANGELOG.md`](https://github.com/facebook/ktfmt/blob/main/CHANGELOG.md)?

**Comment by jsjeon** [comment]
> nit: would you mind adding the change to the [`CHANGELOG.md`](https://github.com/facebook/ktfmt/blob/main/CHANGELOG.md)?

Is it going to be Unreleased / fixed ?

**Review by hick209 — APPROVED** [review]

**Comment by jsjeon** [comment]
Let me know what I need to do regarding Internal Linter error / test failures.

**Comment by hick209** [comment]
@jsjeon it's just a minor thing, don't worry about it, I can take care of it 🙂

**Comment by simonhauck** [comment]
Hi,

Would it be possible for you to create a release with this workaround? I encountered projects where it is now not possible to update the Kotlin version and have proper formatting checks, since this issue is also propagated in the [ktfmt-gralde](https://github.com/cortinico/ktfmt-gradle) plugin.

**Comment by hick209** [comment]
Released on https://github.com/facebook/ktfmt/releases/v0.53

**Pull request closed without merging**